modular_skyrat/modules/dispatching/dispatch_controller.dm

```diff
@@ -0,0 +1,516 @@
+			continue
+		temps[initial(template.name)] = new template()
+	return temps
+
+/datum/controller/subsystem/dispatch/proc/apply_template(mob/user, datum/dispatch_ticket_template/template)
+	ui_data_by_mob[user]["tdata"]["title"] = template.title
+	ui_data_by_mob[user]["tdata"]["extra"] = template.extra
+	ui_data_by_mob[user]["tdata"]["type"] = template.template_type
+	ui_data_by_mob[user]["tdata"]["priority"] = template.template_priority_default
+	ui_data_by_mob[user]["tdata"]["templateName"] = template.name
+
+///Interates through all loaded jobs and assigns them to one or more type caches
+/datum/controller/subsystem/dispatch/proc/load_job_cache()
+	job_cache_roles = list()
+	job_cache_holder = list()
+	for(var/job in SSjob.occupations)
+		if(job in job_cache_roles)
+			continue
+		var/list/roles = list()
+		var/datum/job/job_real = job
+		for(var/ttype in SSDISPATCH_TICKET_TYPES)
+			if(job_real.departments & SSDISPATCH_TICKET_TYPES_DEPT[ttype])
+				roles += ttype
+		job_cache_roles[job_real.title] = roles
+		job_cache_holder[job_real.title] = list()
+
+///Iterates through all joined players and assigns them to their jobs holder cache
+/datum/controller/subsystem/dispatch/proc/load_job_holders()
+	if(!job_cache_roles)
+		CRASH("load_job_holders called before job_cache_roles initialized!")
+
+	for(var/job in job_cache_holder)
+		job_cache_holder[job] = list()
+
+	for(var/key in GLOB.joined_player_list)
+		var/mob/key_mob = get_mob_by_ckey(key)
+		if(!key_mob) continue
+		if(!ishuman(key_mob)) continue
+		if(key_mob.job in job_cache_holder)
+			job_cache_holder[key_mob.job] += key_mob
+
+///Iterates through all job caches and all their holders and assigns them to all applicable type holders
+/datum/controller/subsystem/dispatch/proc/load_type_holders()
+	if(!job_cache_holder)
+		CRASH("load_type_holders called before load_job_holders!")
+
+	job_type_holders = list()
+	for(var/ttype in SSDISPATCH_TICKET_TYPES)
+		job_type_holders[ttype] = list()
+
+	//HOLY FUCK THIS IS UGLY; IF SOMETHING DOESNT WORK CHECK THIS SHIT FIRST
+	for(var/job in job_cache_holder)
+		for(var/mob in job_cache_holder[job])
+			for(var/ttype in SSDISPATCH_TICKET_TYPES)
+				if(ttype in job_cache_roles[job])
+					job_type_holders[ttype] += mob
+
+///Sends a message to every type holder of a specific type, given they are active, unless ignore_active is TRUE
+/datum/controller/subsystem/dispatch/proc/message_type_holders(ticket_type, message, ignore_active=FALSE)
+	for(var/mob/mob in job_type_holders[ticket_type])
+		if((ui_data_by_mob[mob] && ui_data_by_mob[mob]["mdata"]["holderActive"]) || ignore_active)
+			message_holder(mob, message)
+
+///Takes a user, and a list of the ticket data, and constructs a ticket to then alert type holders and slots it into the master list
+/datum/controller/subsystem/dispatch/proc/ticket_create(mob/user, list/tdata)
+	holder_update()
+	var/datum/dispatch_ticket/ticket = new(user, tdata)
+	user.visible_message("[user] makes a loud chime!")
```

Comment:
I feel like this should be more obvious what it is, people won't know what that means. Maybe something like "[user] makes a loud chime - ticket created successfully!"

Reply:
Sure, suggest it so I can web commit it?

Reply:
Sure.